diff --git a/src/llama-chat.cpp b/src/llama-chat.cpp
index 6d822ec..980b4f2 100644
--- a/src/llama-chat.cpp
+++ b/src/llama-chat.cpp
@@ -83,15 +83,27 @@ static const std::map<std::string, llm_chat_template> LLM_CHAT_TEMPLATES = {
 };
 
 llm_chat_template llm_chat_template_from_str(const std::string & name) {
+#ifdef __NAGI__
+    const auto it = LLM_CHAT_TEMPLATES.find(name);
+    return it == LLM_CHAT_TEMPLATES.end() ? LLM_CHAT_TEMPLATE_UNKNOWN : it->second;
+#else
     return LLM_CHAT_TEMPLATES.at(name);
+#endif
 }
 
 llm_chat_template llm_chat_detect_template(const std::string & tmpl) {
+#ifdef __NAGI__
+    const auto it = LLM_CHAT_TEMPLATES.find(tmpl);
+    if (it != LLM_CHAT_TEMPLATES.end()) {
+        return it->second;
+    }
+#else
     try {
         return llm_chat_template_from_str(tmpl);
     } catch (const std::out_of_range &) {
         // ignore
     }
+#endif
 
     auto tmpl_contains = [&tmpl](const char * haystack) -> bool {
         return tmpl.find(haystack) != std::string::npos;
